src/shared_gui_components/Component.cpp

```diff
@@ -464,157 +479,172 @@ void Component::createAbridgedLayout() {
+    tableWidget->setAlternatingRowColors(true);
+    tableWidget->verticalHeader()->hide();
+    tableWidget->horizontalHeader()->hide();
+    tableWidget->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    tableWidget->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    tableWidget->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
+    tableWidget->horizontalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+    tableWidget->horizontalHeader()->setStretchLastSection(true);
+    mainLayout->addWidget(tableWidget);
+    return tableWidget;
+  };
+
+  auto addRowToTableWidget = [](QTableWidget* tableWidget, const QString& header, const QString& value) {
     tableWidget->insertRow(tableWidget->rowCount());
 
-    auto* item = new QTableWidgetItem(attribute.name().c_str());
+    auto* item = new QTableWidgetItem(header);
     tableWidget->setItem(tableWidget->rowCount() - 1, 0, item);
 
-    boost::optional<std::string> optionalUnits = attribute.units();
+    item = new QTableWidgetItem(value);
+    tableWidget->setItem(tableWidget->rowCount() - 1, 1, item);
+  };
 
-    openstudio::AttributeValueType type = attribute.valueType();
+  auto makeTableShowCompletely = [](QTableWidget* tableWidget) {
+    // make the table show completely
+    const int rowHeight = tableWidget->rowHeight(0);
+    const int rowCount = tableWidget->rowCount();
+    const int tableHeight = rowHeight * rowCount;
+    tableWidget->setFixedHeight(tableHeight);
+  };
+
+  //if (!m_attributes.empty()){
+  {
+    ///! Attributes
+    ///! Class BCL only stores double (optional units),
+    ///! int (optional units), and string, with their names.
+    auto* label = new QLabel("Attributes");
+    label->setObjectName("H1");
+    mainLayout->addWidget(label);
 
-    if (type == AttributeValueType::Boolean) {
-      bool success = attribute.valueAsBoolean();
-      if (success) {
-        string = "true";
+    auto* tableWidget = createAndRegisterTableWidgetWithTwoColums();
+    for (const Attribute& attribute : m_attributes) {
+
+      boost::optional<std::string> optionalUnits = attribute.units();
+
+      const openstudio::AttributeValueType type = attribute.valueType();
+
+      QString display;
+      if (type == AttributeValueType::Boolean) {
+        display = attribute.valueAsBoolean() ? "true" : "false";
+      } else if (type == AttributeValueType::Double) {
+        display.setNum(attribute.valueAsDouble());
+      } else if (type == AttributeValueType::Integer) {
+        display.setNum(attribute.valueAsInteger());
+      } else if (type == AttributeValueType::Unsigned) {
+        display.setNum(attribute.valueAsUnsigned());
+      } else if (type == AttributeValueType::String) {
+        display = attribute.valueAsString().c_str();
+      } else if (type == AttributeValueType::AttributeVector) {
+        AttributeVector attributeVector = attribute.valueAsAttributeVector();
+        // TODO handle this case
```

Comment:
Looks like `attribute.toString` might work for the vector case, might be kind of ugly but I think this is rare.

Reply:
It wasn't there before and no one complained, so I didn't change it :)

toJSONString would be an optional, or just managed it by looping.

Reply:
Yeah, I kind of doubt anyone is using AttributeVector